awx/main/utils/common.py

```diff
@@ -887,30 +888,38 @@ def get_current_apps():
     return current_apps
 
 
-def get_custom_venv_choices(custom_paths=None):
+def get_custom_venv_choices():
     from django.conf import settings
 
-    custom_paths = custom_paths or settings.CUSTOM_VENV_PATHS
-    all_venv_paths = [settings.BASE_VENV_PATH] + custom_paths
+    all_venv_paths = []
+
+    all_venv_paths.append(settings.BASE_VENV_PATH)  # get paths from settings
+
+    for root, dir, files in os.walk('../awx_devel/var/lib/awx/venv/'):  # get paths on machine
+        all_venv_paths.append(root)
+
     custom_venv_choices = []
 
-    for custom_venv_path in all_venv_paths:
+    for venv_path in all_venv_paths:
         try:
-            if os.path.exists(custom_venv_path):
+            if os.path.exists(venv_path):
                 custom_venv_choices.extend(
-                    [
-                        os.path.join(custom_venv_path, x, '')
-                        for x in os.listdir(custom_venv_path)
-                        if x != 'awx'
-                        and os.path.isdir(os.path.join(custom_venv_path, x))
-                        and os.path.exists(os.path.join(custom_venv_path, x, 'bin', 'activate'))
-                    ]
+                    [os.path.join(venv_path.replace('..', ''), x, '') for x in os.listdir(venv_path) if os.path.exists(os.path.join(venv_path, x, 'activate'))]
                 )
         except Exception:
             logger.exception("Encountered an error while discovering custom virtual environments.")
     return custom_venv_choices
 
 
+def get_custom_venv_pip_freeze(venv_path):
+    try:
+        freeze_data = subprocess.run([f"{venv_path}/pip", "freeze"], capture_output=True)
+        pip_data = (freeze_data.stdout).decode('ascii')
```

Comment:
I'm a bit uncomfortable with assuming they are guaranteed to be ASCII.

Reply:
I tried looking around to see if the output of `pip freeze` would ever contain non-ASCII characters but don't see anything.  The closest I see is that pip supports a UTF-8 encoding for requirements files: https://pip.pypa.io/en/stable/cli/pip_install/#requirements-file-format

Reply:
subprocess outputs to byte format by default, so I have to decode. I picked ascii arbitrarily so I'm not particularly tied to it, that being said, I _highly_ doubt non-ascii characters would ever be present. 
I could decode to utf-8 if we think it's ever going to be problematic

Reply:
yeah, hard to say.  I would mostly agree, but haven't come up with any hard reason why a package couldn't be named something with some non-ascii characters.

Reply:
will there ever be any `'` 😂 cause then we've got trouble.

Reply:
swapped to utf-8